### PR TITLE
[2-dev] Update min SDK; add Angular analyzer plugin, etc

### PR DESCRIPTION
**Pull request by chalin**
General changes:
- Set minimal SDK to 2.0.0-dev.28.0
- Dart 2 changes: renamed variables like, `JSON`, `PI`, etc. to `json`, `pi`, etc.

Web-angular template changes:

- Enabled Angular analyzer plugin
- Replace builder config file with new named ones.
- Switch to mock `pageloader`
- `CORE_DIRECTIVES` -> `coreDirectives`
- Dropped <del>`const`</del> qualifier from literals used used in metadata annotations
- Switched to using new provider classes like `ClassProvider`

### CODE REVIEW

**Comment by matanlurey** [comment]
> Dropped <del>`const`</del> qualifier from literals used used in metadata annotations

@chalin: This works?

**Comment by chalin** [comment]
> This works?

Yes! :). The web-template builds and tests pass:

```terminal
web-angular > pub run build_runner test --fail-on-severe --delete-conflicting-outputs --output=build -- -p chrome --reporter=expanded
[INFO] Generating build script completed, took 230ms
[INFO] Reading cached asset graph completed, took 1.1s
[INFO] Checking for updates since last build completed, took 443ms
[INFO] Running build completed, took 11ms
[INFO] Caching finalized dependency graph completed, took 280ms
[INFO] Reading manifest at build/.build.manifest completed, took 2ms
[INFO] Deleting previous outputs in `build` completed, took 919ms
[INFO] Creating merged output dir `build` completed, took 5.2s
[INFO] Writing asset manifest completed, took 3ms
[INFO] Succeeded after 6.5s with 0 outputs
Running tests...

00:00 +0: compiling test/app_test.dart
00:00 +0: test/app_test.dart: title
00:00 +1: All tests passed!
```

File: test/cli_test.dart [hunk 1]
```diff
@@ -63,7 +63,7 @@ void main() {
     test('machine format', () {
       return app.process(['--machine']).then((_) {
         _expectOk();
-        List results = JSON.decode(logger.getStdout());
+        List results = json.decode(logger.getStdout());
```
Comment (kwalrath) [inline test/cli_test.dart]: @kevmoo should we be using [jsonDecode](https://api.dartlang.org/dev/dart-convert/jsonDecode.html)?

Comment (chalin) [reply test/cli_test.dart]: Changed

**Comment by chalin** [comment]
This good to merge (not publish, but merge into 2-dev)?

**Pull request merged**